backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/index/impl/IndexManagerBackendContext.java

```diff
@@ -250,4 +261,57 @@ String toElasticsearchId(String tenantId, String id) {
 	public ElasticsearchAnalysisPerformer createAnalysisPerformer(ElasticsearchIndexModel model) {
 		return new ElasticsearchAnalysisPerformer( model, link.getWorkFactory(), generalPurposeOrchestrator );
 	}
+
+	public void start(BackendStartContext context) {
+		indexLayoutStrategyHolder = createIndexLayoutStrategy( context );
+		typeNameMapping.onStart( indexLayoutStrategyHolder.get() );
+		searchProjectionBackendContext = new SearchProjectionBackendContext(
+				typeNameMapping.getTypeNameExtractionHelper(),
+				multiTenancyStrategy.idProjectionExtractionHelper()
+		);
+	}
+
+	public void stop() {
+		try ( Closer<RuntimeException> closer = new Closer<>() ) {
+			closer.push( BeanHolder::close, indexLayoutStrategyHolder );
+		}
+	}
```

Comment:
Would it make sense to shove the state into `ElasticsearchLink` instead? That would at least limit the number of classes that keep state around...

Reply:
something like I've done in the second commit?